gpt_engineer/project.py

```diff
@@ -0,0 +1,16 @@
+from gpt_engineer.db import DBs
+
+
+class Project:
+    def __init__(self, dbs: DBs):
+        self.dbs = dbs
+
+    def update(self, diff):
+        """Updates project files based on the diff"""
+        # TODO: how do we represent diff?
+        raise NotImplementedError
+
+    def run_tests(self):
```

Comment:
I like the execution env stuff that are like "run_test" and "run_entrypoint" being separate methods/functions though!

Can't decide on this vs steps, will have to sleep on it:)